pkg/apiserver/handlers.go

```diff
@@ -64,8 +64,8 @@ func ReadOnly(handler http.Handler) http.Handler {
 			handler.ServeHTTP(w, req)
 			return
 		}
-		w.WriteHeader(http.StatusForbidden)
-		fmt.Fprintf(w, "This is a read-only endpoint.")
+		w.Header().Set("Allow", "GET")
+		http.Error(w, "This is a read-only endpoint.", http.StatusMethodNotAllowed)
```

Comment:
I see a few other places in this file where we do the same thing (return a plain text response instead of a json status object). Feel free to update them all while you are it :)
Ref https://github.com/GoogleCloudPlatform/kubernetes/issues/1990